packages/client/src/open-feature.ts

```diff
@@ -196,19 +207,25 @@ export class OpenFeatureAPI extends OpenFeatureCommonAPI<Provider, Hook> impleme
     oldContext: EvaluationContext,
     newContext: EvaluationContext,
   ): Promise<void> {
-      const providerName = provider.metadata.name;
-      return provider.onContextChange?.(oldContext, newContext).then(() => {
-        this.getAssociatedEventEmitters(clientName).forEach((emitter) => {
-          emitter?.emit(ProviderEvents.ContextChanged, { clientName, providerName });
-        });
-        this._events?.emit(ProviderEvents.ContextChanged, { clientName, providerName });
-      }).catch((err) => {
-        this._logger?.error(`Error running ${provider.metadata.name}'s context change handler:`, err);
-        this.getAssociatedEventEmitters(clientName).forEach((emitter) => {
-          emitter?.emit(ProviderEvents.Error, { clientName, providerName, message: err?.message, });
-        });
-        this._events?.emit(ProviderEvents.Error, { clientName, providerName, message: err?.message, });
+    const providerName = provider.metadata.name;
+    try {
+      await provider.onContextChange?.(oldContext, newContext);
```

Comment:
This was the bug - the previous implementation only emitted events if the provider had a `onContextChange` implementation, which is optional, because it was chained like this:

`provider.onContextChange?.(oldContext, newContext).then(...`

Now we await the `onContextChange` OR just run the handlers.